src/resolver/resolve.cc

```diff
@@ -14,15 +14,32 @@
 
 namespace flox::resolver {
 
+/* -------------------------------------------------------------------------- */
+
+void
+Resolved::Input::limitLocked()
+{
+  auto type = this->locked.at( "type" ).get<std::string>();
+  if ( ! ( ( type == "path" ) || ( type == "tarball" ) || ( type == "file" ) ) )
+    {
+      this->locked.erase( "narHash" );
+    }
+  this->locked.erase( "lastModified" );
+  this->locked.erase( "lastModifiedDate" );
+  this->locked.erase( "revCount" );
+  this->locked.erase( "shortRev" );
```

Comment:
question: curious to understand why some of these fields make things impure - makes sense why lastModified might change, but why does shortRev?

Reply:
They don't they're just spurious.
We only need `rev` for `git`_like_ fetchers ( `git`, `github`, `gitlab`, `mercurial` ), or `narHash` for `file`_like_ fetchers ( `path`, `file`, `tarball` ).

The extra fields aren't useful to us for the purposes of reproducing an env so I dropped them to cut down on the size of the lock.

Reply:
Thanks